src/st_lib.c

```diff
@@ -132,17 +131,30 @@ void STlib_drawNum
     num = -num;
   }
 
+  for (i = 1; i <= 9; ++i)
```

Comment:
It shouldn't be necessary to recalculate this over and over again during each function call. Let's store the result in static variables, please.

Reply:
But we use `STlib_drawNum` for big and small fonts. I guess we can add fields to `st_number_t` or just redraw STBAR every tic.

Reply:
I simplified it here: https://github.com/fabiangreffrath/woof/pull/691/commits/54deef1f0abba0a4f7d5dbed5c036eb2702a20df No `oldnum` optimization, but we didn't have it until recently.